File: .snippets/code/chain-interactions/query-data/runtime-api-calls/papi/runtime-apis.ts
import { createClient } from 'polkadot-api';
import { getWsProvider } from 'polkadot-api/ws-provider/node';
import { withPolkadotSdkCompat } from 'polkadot-api/polkadot-sdk-compat';
import { polkadotTestNet } from '@polkadot-api/descriptors';

const POLKADOT_TESTNET_RPC = 'INSERT_WS_ENDPOINT';

// Example address to query
const ADDRESS = 'INSERT_ADDRESS';

async function main() {
  // Create the client connection
  const client = createClient(
    withPolkadotSdkCompat(getWsProvider(POLKADOT_TESTNET_RPC))
  );

  // Get the typed API for Polkadot Hub TestNet (Paseo Asset Hub)
  const api = client.getTypedApi(polkadotTestNet);

  console.log('Connected to Polkadot Hub TestNet');
  console.log(`Querying runtime APIs for: ${ADDRESS}\n`);

  // Call AccountNonceApi to get the account nonce
  const nonce = await api.apis.AccountNonceApi.account_nonce(ADDRESS);
  console.log('AccountNonceApi Results:');
  console.log(`  Account Nonce: ${nonce}`);

  // Query metadata versions using Metadata runtime API
  const metadataVersions = await api.apis.Metadata.metadata_versions();
  console.log('\nMetadata API Results:');
  console.log(`  Supported Metadata Versions: [${metadataVersions.join(', ')}]`);

  // Disconnect the client
  client.destroy();
}

main().catch(console.error);

File: .snippets/code/chain-interactions/query-data/runtime-api-calls/psi/runtime_apis.py
from substrateinterface import SubstrateInterface

ASSET_HUB_RPC = "INSERT_WS_ENDPOINT"

# Example address to query
ADDRESS = "INSERT_ADDRESS"


def main():
    # Connect to Polkadot Hub TestNet (Paseo Asset Hub)
    substrate = SubstrateInterface(url=ASSET_HUB_RPC)

    print("Connected to Polkadot Hub TestNet")
    print(f"Querying runtime APIs for: {ADDRESS}\n")

    # Call AccountNonceApi to get the account nonce
    nonce = substrate.runtime_call("AccountNonceApi", "account_nonce", [ADDRESS])
    print("AccountNonceApi Results:")
    print(f"  Account Nonce: {nonce.value}")

    # Query runtime version using Core runtime API
    version = substrate.runtime_call("Core", "version", [])
    print("\nCore API Results:")
    print(f"  Spec Name: {version.value['spec_name']}")
    print(f"  Spec Version: {version.value['spec_version']}")
    print(f"  Impl Version: {version.value['impl_version']}")

    # Close connection
    substrate.close()


if __name__ == "__main__":
    main()
